libs/langchain/langchain/vectorstores/opensearch_vector_search.py

```diff
@@ -633,13 +658,24 @@ def _raw_similarity_search_with_score(
                     boolean_filter = filter
 
             if boolean_filter != {}:
-                search_query = _approximate_search_query_with_boolean_filter(
-                    embedding,
-                    boolean_filter,
-                    k=k,
-                    vector_field=vector_field,
-                    subquery_clause=subquery_clause,
-                )
+                if text_field != {}:
```

Comment:
the default value for this in L625 is a string but here we're checking that it's not an empty dict — is that intentional / correct?

Reply:
perhaps would make sense to make default value for text_field None and then here we check if `text_field is not None`

Reply:
Oh true I will fix the text_field type as a string